adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -196,39 +197,66 @@ func (b *BM25Searcher) wand(
 	averagePropLength = averagePropLength / float64(len(properties))
 
 	// preallocate the results (+1 is for full query)
-	results := make(terms, 0, len(queryTextTerms)+len(queryStringTerms)+1)
-	indices := make([]map[uint64]int, 0, len(queryTextTerms)+len(queryStringTerms)+1)
+	fullQueryLength := 0
+	if len(propertyNamesFullQuery) > 0 {
+		fullQueryLength = 1
+	}
+	stringLength := 0
+	if len(propertyNamesString) > 0 {
+		stringLength = len(queryStringTerms)
+	}
+	textLength := 0
+	if len(propertyNamesText) > 0 {
+		textLength = len(queryTextTerms)
+	}
+	results := make(terms, textLength+stringLength+fullQueryLength)
+	indices := make([]map[uint64]int, textLength+stringLength+fullQueryLength)
+	createErrors := make([]error, textLength+stringLength+fullQueryLength)
+
+	var wg sync.WaitGroup
 
 	if len(propertyNamesText) > 0 {
+		wg.Add(len(queryTextTerms))
 		for i, queryTerm := range queryTextTerms {
-			termResult, docIndices, err := b.createTerm(N, filterDocIds, queryTerm, propertyNamesText, propertyBoosts, duplicateTextBoost[i])
-			if err != nil {
-				return nil, nil, err
-			}
-			results = append(results, termResult)
-			indices = append(indices, docIndices)
+			go func(j int, queryTerm string) {
```

Comment:
using `errgroup.Group` instead of `sync.WaitGroup` in all these cases would be more readable, since we are just returning the first error anyway

Reply:
I did not know that, thanks!